libs/full/runtime_distributed/src/server/runtime_support_server.cpp

```diff
@@ -227,8 +226,10 @@ namespace hpx { namespace components { namespace server {
     void runtime_support::dijkstra_make_black()
     {
         // Rule 1: A machine sending a message makes itself black.
-        std::lock_guard<dijkstra_mtx_type> l(dijkstra_mtx_);
-        dijkstra_color_ = true;
+        if (!dijkstra_color_)
+        {
+            dijkstra_color_ = true;
+        }
```

Comment:
I do not have enough confidence in the `compare_exchange` implementation. Sometimes when no modification is supposed to be made, I find it can be more expensive than a simple read?

Reply:
I think directly store is more expensive than read as it will always invalidate the cache line.